### ISSUE TITLE
Question. how to sort by the most matches on a collection

### ISSUE THREAD

**Issue by ohpoloha** [querydsl/querydsl#2203]
say I have an two tables, project, tag and project_tag. 

project entity has a collection of tags which stores project_tag. I want to search for the project that has certain tag_id. For example, project has tag_id 1, 2 or 3, so I do `QProject.project.tags.any().tagId.in(1,2,3);`

This does exactly what I want! Then I want to sort by project that has the most matches. Meaning project which has all 3 matches (1, 2 and 3) would be in the top. then 2 matches and finally one match.

basically the same as [here](https://stackoverflow.com/questions/18741592/how-to-in-mysql-order-rows-by-number-of-matches-in-associative-table) but in querydsl
